src/main/java/com/example/springbootmongodb/service/JokeService.java

```diff
@@ -0,0 +1,29 @@
+package com.example.springbootmongodb.service;
+
+import com.example.springbootmongodb.app.model.Joke;
+import com.example.springbootmongodb.db.JokeRepository;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Optional;
+
+@Service
+public class JokeService {
+    public static final String URL = "https://official-joke-api.appspot.com/random_joke";
+    private final RestTemplate restTemplate;
+    private final JokeRepository jokeRepository;
+
+    public JokeService(RestTemplateBuilder restTemplateBuilder, JokeRepository jokeRepository) {
+        this.restTemplate = restTemplateBuilder.build();
+        this.jokeRepository = jokeRepository;
+    }
+
+    public Optional<Joke> getJoke() {
+        Joke joke = this.restTemplate.getForObject(URL, Joke.class);
```

Comment:
when using Optional, you should not mix it with traditional null checks etc. I would expect something like 
```
return Optional
    .ofNullable(restTemplate.get())
    .map(this::saveIfNotExist);
```

Reply:
Refactored to not use null check, only Optional.